apis/python/tests/test_dataframe.py

```diff
@@ -461,6 +468,229 @@ def test_bool_enums(tmp_path, data_and_expected_levels):
         assert actual == expect
 
 
+@pytest.mark.parametrize("extend_not_write", [False, True])
+def test_extend_enumeration_values(tmp_path, extend_not_write):
+    """Compares the older create+write path against create+extend path"""
+    uri = tmp_path.as_posix()
+    domain = [[0, 2]]
+
+    schema = pa.schema(
```

Comment:
Tests are missing cases of extending an _ordered_ dict.  While the current code base doesn't branch based upon this, we should still test it.

Reply:
In particular, in the dedup case it is super important that the dedup preserve ordering....

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/3815/commits/6469953614136b2745cc963afe1c15c7f0991e9c